compiler/qsc_rir/src/passes/remap_block_ids.rs

```diff
@@ -54,6 +54,7 @@ pub fn remap_block_ids(program: &mut Program) {
             // the same blocks back-to-back.
             continue;
         }
+        blocks_to_visit.retain(|id| !successors.contains(id));
```

Comment:
A comment here would be useful. Something like: "Since we are going to extend the blocks to visit using the successors of the block we are currently visiting, remove the successor blocks already in the blocks to visit.".

Reply:
Added comment!